libs/velo-external-db-core/src/service/schema.spec.ts

```diff
@@ -71,20 +72,235 @@ describe('Schema Service', () => {
+            const collectionCapabilities = {
+                dataOperations: [],
+                fieldTypes: [],
+                collectionOperations: [],
+            }
+            
+            driver.givenAllSchemaOperations()
+            driver.givenCollectionCapabilities(collectionCapabilities)
+            driver.givenColumnCapabilities()
+            driver.givenListResult(ctx.dbsWithIdColumn)
+            
+
+            await expect( env.schemaService.listCollections([]) ).resolves.toEqual( collectionsListFor(ctx.dbsWithIdColumn, collectionCapabilities))
+        })
+
+        test('create new collection without fields', async() => {
+            driver.givenAllSchemaOperations()
+            driver.expectCreateOf(ctx.collectionName)
+            schema.expectSchemaRefresh()
+
+            await expect(env.schemaService.createCollection({ id: ctx.collectionName, fields: [] })).resolves.toEqual({
+                collection: { id: ctx.collectionName, fields: [] } 
+            })
+        })
+
+        test('create new collection with fields', async() => {
+            const fields = [{
+                key: ctx.column.name,
+                type: convertFieldTypeToEnum(ctx.column.type),
+            }]
+            driver.givenAllSchemaOperations()
+            schema.expectSchemaRefresh()            
+            driver.expectCreateWithFieldsOf(ctx.collectionName, fields)
+    
+            await expect(env.schemaService.createCollection({ id: ctx.collectionName, fields })).resolves.toEqual({
+                collection: { id: ctx.collectionName, fields }
+            })
+        })
+
+        test('compareColumnsInDbAndRequest function - add columns', async() => {
+            const { compareColumnsInDbAndRequest } = env.schemaService
+            const columnsInDb = [{
+                field: ctx.column.name,
+                type: ctx.column.type
+            }]
+            const columnsInRequest = [{
+                key: ctx.column.name,
+                type: convertFieldTypeToEnum(ctx.column.type),
+            }]
+            const newColumn = {
+                key: ctx.anotherColumn.name,
+                type: convertFieldTypeToEnum(ctx.anotherColumn.type)
+            }
+            expect(compareColumnsInDbAndRequest([], []).columnsToAdd).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, columnsInRequest).columnsToAdd).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, []).columnsToAdd).toEqual([])
+            expect(compareColumnsInDbAndRequest([], columnsInRequest).columnsToAdd).toEqual(convertWixFormatFieldsToInputFields(columnsInRequest))
+            expect(compareColumnsInDbAndRequest(columnsInDb, [...columnsInRequest, newColumn]).columnsToAdd).toEqual(convertWixFormatFieldsToInputFields([newColumn]))
+        })
+
+        test('compareColumnsInDbAndRequest function - remove columns', async() => {
```

Comment:
Under the `describe('Collection new SPI', () => {`
add another describe for all the private methods and converters.
It will look neater.